core-bundle/src/Twig/Loader/ContaoFilesystemLoader.php

```diff
@@ -367,34 +368,13 @@ public function buildInheritanceChains(): void
         $this->inheritanceChains = $hierarchy;
     }
 
-    /**
-     * Split a Contao name into [namespace, short name]. The short name part
-     * will be null if $name is only a namespace.
-     *
-     * If parsing fails - i.e. if the given name does not describe a "Contao"
-     * or "Contao_*" namespace - null is returned instead.
-     */
-    private function parseContaoName(string $logicalNameOrNamespace): ?array
-    {
-        if (1 === preg_match('%^@(Contao(?:_[a-zA-Z0-9_-]+)?)(?:/(.*))?$%', $logicalNameOrNamespace, $matches)) {
-            return [$matches[1], $matches[2] ?? null];
-        }
-
-        return null;
-    }
-
-    private function getIdentifier(string $name): string
-    {
-        return preg_replace('%(?:.*/)?(.*)(\.html5|\.html.twig)%', '$1', $name);
-    }
-
     /**
      * Returns the template name of a theme specific variant of the given name
      * or null if not applicable.
      */
     private function getThemeTemplateName(string $name): ?string
     {
-        if (null === ($parts = $this->parseContaoName($name)) || 'Contao' !== $parts[0]) {
+        if (null === ($parts = ContaoTwigUtil::parseContaoName($name)) || 'Contao' !== $parts[0]) {
```

Comment:
Why this check? If `$parts` is not `null`, `$parts[0]` will always be `Contao`, won't it?

Reply:
see https://github.com/contao/contao/pull/3201/files#r678459280